controllers/util/mw_util.go

```diff
@@ -200,16 +201,76 @@ func (mwu *MWUtil) generateVRGManifest(
 	})
 }
 
-func (mwu *MWUtil) CreateOrUpdateVRGRolesManifestWork(namespace string) error {
-	manifestWork, err := mwu.generateVRGRolesManifestWork(namespace)
+func (mwu MWUtil) ClusterRolesList(clusterNames *sets.String) error {
+	manifestworks := &ocmworkv1.ManifestWorkList{}
+	if err := mwu.Client.List(mwu.Ctx, manifestworks); err != nil {
+		return fmt.Errorf("manifestworks list: %w", err)
+	}
+
+	for i := range manifestworks.Items {
+		manifestwork := &manifestworks.Items[i]
+		if manifestwork.ObjectMeta.Name == ClusterRolesManifestWorkName {
+			*clusterNames = clusterNames.Insert(manifestwork.ObjectMeta.Namespace)
```

Comment:
What the heck this is doing?  I can wrap my head around it.  It inserting a cluster name (which is MW namespace) into a map of strings but the return value of the Insert is a string which is assigned to *clusterNames???

Reply:
It outputs a set of namespace names, i.e. cluster names, that have a manifest work named "ramendr-roles".  It uses kubernetes' `sets.String` which is a set of strings, implemented as a map with string keys and empty values.

Reply:
Okay. So I was looking at this `Insert`.  So you are not using that one then??? I can double-check, I am just being lazy.
https://github.com/kubernetes/apimachinery/blob/v0.22.0/pkg/util/sets/string.go#L49

Reply:
Ah got it.  The confusion is the name `String` through me off.  I thought it is a simple string, but it is typedefed to a map.

Reply:
This routine is currently only used by unit test to confirm that only expected cluster roles manifestworks exist.